src/test/java/net/minecraftforge/debug/world/MultiNoiseBiomeSourceLoadingEventTest.java

```diff
@@ -0,0 +1,259 @@
+        ChunkGenerator namelessGenerator = new NoiseBasedChunkGenerator(noiseParameters, namelessSource, 0, () -> overworldNoiseSettings);
+        LevelStem namelessDimension = new LevelStem(() -> overworldDimensionType, namelessGenerator);
+        
+        // now generate those dimensions
+        DataGenerator generator = event.getGenerator();
+        Path resourcesFolder = generator.getOutputFolder();
+        Gson gson = new GsonBuilder()
+            .setPrettyPrinting()
+            .create();
+        
+        DataProvider provider = new DataProvider()
+        {
+            @Override
+            public void run(HashCache cache) throws IOException
+            {
+                BiConsumer<ResourceLocation, LevelStem> dimensionWriter = (id, dimension)->
+                {
+                    JsonElement json = LevelStem.CODEC.encodeStart(writeOps, dimension).get().orThrow();
+                    Path path = resourcesFolder.resolve(String.join("/", PackType.SERVER_DATA.getDirectory(), id.getNamespace(), "dimension", id.getPath()) + ".json");
+                    try
+                    {
+                        DataProvider.save(gson, cache, json, path);
+                    }
+                    catch (IOException e)
+                    {
+                        e.printStackTrace();
+                    }
+                };
+                dimensionWriter.accept(PRESET_TEST, presetDimension);
+                dimensionWriter.accept(NAMED_TEST, namedDimension);
+                dimensionWriter.accept(NAMELESS_TEST, namelessDimension);
+            }
+
+            @Override
+            public String getName()
+            {
+                return MODID + " worldgen data provider";
+            }
+        };
+        generator.addProvider(provider);
+    }
+    
+    private void onLoadBiomeSource(MultiNoiseBiomeSourceLoadingEvent event)
+    {
+        // we want to test the following things:
+        // * the dimensions that should have extra biomes added have extra biomes added
+        // * the dimensions that did not have extra biomes added do not have extra biomes added
+        // * extra biomes are not preserved in the biome source across multiple world creates/joins
+        
+        // expected behaviour is that overworld and overworld preset json dimension have an extra end biome,
+        // nether has an extra jungle biome,
+        // and the named test dimension has an extra nether biome
+        if (Objects.equals(event.getName(), Preset.OVERWORLD.name))
+        {
+            Biome end = event.getRegistries().registryOrThrow(Registry.BIOME_REGISTRY).get(Biomes.SMALL_END_ISLANDS);
+            var parameters = event.getParameters();
+            LOGGER.info("Adding end islands biome to overworld biome source with {} biomes", parameters.size());
+            parameters.add(Pair.of(new ParameterPoint(
```

Comment:
I wonder if it would be worth it to add a `addParameter(ParameterPoint, Supplier<Biome>)` method to the event, for modder convenience.

Reply:
Fair enough, added in next push